tls-support-plan.md

```diff
@@ -0,0 +1,81 @@
+## TLS Support for Database Handlers
+
+### General approach
++ Extract sslmode params from credentials into backend connection options
+  + sslmode: Provide different levels of protection.
+  + sslca: The Certificate Authority (CA) certificate file. This option, if used, must specify the same certificate used by the server
+  + sslcert: The client public key certificate file.
+  + sslkey: The client private key file.
+  + sslcrl: The certificates revoked by certificate authorities.
++ Dial backend to get net.Conn
++ Default to secure connection if TLS supported
++ Ensure client TLS requirements match server requirements and support, FAIL if not.
+  + Default to insecure connection, regardless of support
+  + Use TLS if server demands it
++ Upgrade net.Conn to tls.Conn and use appropriate strategy for sslmode
```

Comment:
this step seems like it could be the bulk of the work. is it worth describing in more detail? or is what is described in the handler-specific sections below sufficient?

Reply:
I think the handler specific stuff below is sufficient for describing the handshakes. configuring TLS using the appropriate strategy is Golang thing more than anything else